### ISSUE TITLE
migrate adapter from 0.169 to 0.176

### ISSUE THREAD

**Issue by matthieularere-msq** [prebid/prebid-server#1998]
Hi,

I was working on a bidder adapter in july, it's working fine with release 0.169.0. However when upgrading to 0.176.0 it no longer works and I have not much clue about why. I think it's ignored because while it's in my payload calling endpoint /openrtb2/auction in the response I can not see it in ext.debug.httpcalls.
Though, it seems to be recognized by prebid-server as I can call setuid?bidder=<bidder_name>&gdpr=1&gdpr_consent=<consent_string> and get my cookie uids updated (and I don't see the "The bidder name provided is not supported by Prebid Server" message).

How could I check the reasons why an auction ignore one of the bidders provided ?

Thanks

**Comment by SyntaxNode** [comment]
Hi @matthieularere-msq. Version 0.173.0 included major changes to adapter user syncing. I wrote updated docs, but they're still in review and haven't merged into docs.prebid.org yet. You can access the [updated document here](https://github.com/prebid/prebid.github.io/blob/9dd5a8be71ce59701732aef3b82937144beb48a5/prebid-server/developers/add-new-bidder-go.md#bidder-info) and follow examples from other adapters. If you have any questions, you're welcomed to post them to this thread.

**Comment by matthieularere-msq** [comment]
Hi @SyntaxNode . I saw that indeed and I have made the changes for this so it's theorically ok. What struggles me is more about the fact that my openrtb requests including my new adapter seems to ignore it. Is there a way to activate some log that would explain me why in my openrtb.imp[0].ext this new bidder is not called ?

**Comment by matthieularere-msq** [comment]
I can see my `Builder` method is called when prebid-server starts so I guess its registered with the core. I double checked that I added the same things in `openrtb_ext/bidders.go` and `exchange/adapter_builders.go`

Indeed I can't see much information in logs. All I can tell is the even thought my bidder Builder what loaded by prebid-server, when I sent an auction with it I can't see neither `MakeRequests` nor  `MakeBids` called. I guessed it might been about gdpr so I changed my vendorId to the appnexus one (because appnexus is successfully called) but it does not change anything.

**Comment by SyntaxNode** [comment]
You might want to check the json schema next. It's going to be difficult to guide you to a solution without seeing the code. If you would post a draft PR I could provide more specific suggestions.

**Comment by matthieularere-msq** [comment]
I've added a few Println to check what happens during a request and finaly it's gdpr related, even thought I am using a vendorId that's works fine for another bidder. I guess it's not working the way I did expect.

in `gdpr/impl.go`, call to `AuctionActivitiesAllowed` returns `p.defaultVendorPermissions` which equals to `false false false <nil>`
`p.vendorIDs` does not contain my bidder. I've dig a little bit to find out where p.vendorIDs is set but it's a bit a treasure hunting in the source code. 

There was an update in gdpr-go with release 0.176.0, I wonder if that's what introduce the issue I face as I have no gdpr issue in 0.169.0. I'll check tomorrow weither my bidder works with 0.175.0.
